app/screens/post_priority_picker/post_priority_picker.tsx

```diff
@@ -0,0 +1,258 @@
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {View} from 'react-native';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import FormattedText from '@components/formatted_text';
+import {Screens} from '@constants';
+import {PostPriorityColors, PostPriorityType} from '@constants/post';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import useNavButtonPressed from '@hooks/navigation_button_pressed';
+import BottomSheet from '@screens/bottom_sheet';
+import {dismissBottomSheet} from '@screens/navigation';
+import {bottomSheetSnapPoint} from '@utils/helpers';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import Footer from './footer';
+import PickerOption from './picker_option';
+
+import type {BottomSheetFooterProps} from '@gorhom/bottom-sheet';
+
+export const POST_PRIORITY_PICKER_BUTTON = 'close-post-priority-picker';
+
+type Props = {
+    componentId: string;
+    isPostAcknowledgementEnabled: boolean;
+    isPersistenNotificationsEnabled: boolean;
+    postPriority: PostPriority;
+    updatePostPriority: (data: PostPriority) => void;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        backgroundColor: theme.centerChannelBg,
+    },
+    titleContainer: {
+        alignItems: 'center',
+        flexDirection: 'row',
+    },
+    title: {
+        color: theme.centerChannelColor,
+        ...typography('Heading', 600, 'SemiBold'),
+    },
+    betaContainer: {
+        backgroundColor: PostPriorityColors.IMPORTANT,
+        borderRadius: 4,
+        paddingHorizontal: 4,
+        marginLeft: 8,
+    },
+    beta: {
+        color: '#fff',
+        ...typography('Body', 25, 'SemiBold'),
+    },
+
+    optionsContainer: {
+        paddingTop: 12,
+    },
+
+    optionsSeparator: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        height: 1,
+    },
+
+    toggleOptionContainer: {
+        marginTop: 16,
+    },
+}));
+
+const PostPriorityPicker = ({
+    componentId, isPostAcknowledgementEnabled, isPersistenNotificationsEnabled,
+    postPriority, updatePostPriority,
+}: Props) => {
+    const {bottom} = useSafeAreaInsets();
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+
+    const style = getStyleSheet(theme);
+
+    const close = useCallback(() => {
+        return dismissBottomSheet(Screens.POST_PRIORITY_PICKER);
+    }, []);
+
+    useNavButtonPressed(POST_PRIORITY_PICKER_BUTTON, componentId, close, []);
+
+    const [data, setData] = useState<PostPriority>(postPriority);
+
+    const displayPersistentNotifications = isPersistenNotificationsEnabled && data.priority === PostPriorityType.URGENT;
+
+    const snapPoints = useMemo(() => {
+        let COMPONENT_HEIGHT = 280;
```

Comment:
how was this measured ?

Reply:
By measuring individual component heights.